pkg/config/config_template.yaml

```diff
@@ -127,6 +127,20 @@ api_key:
 # number of check runners instances concurrently. Setting the value to 1
 # would result in checks running sequentially.
 #
+# NOTE: due to the nature of the python interpreter and its global interpreter
+# lock (GIL) only one native thread may be running on the python interpreter at
+# a time. Concurrent check runners imply concurrently executing checks must contend
+# for the GIL. This has the side-effect of increasing wall-time execution times for
+# the checks. The aggregate system/user-time for all checks however, should be
+# decreased. Concurrency works particularly well with I/O bound tasks (or at least
+# IO intensive) as the GIL will typically be released while waiting for IO, allowing
+# for other checks to grab a hold of the GIL and continue execution. For CPU bound
+# checks with a low IO activity, it becomes increasingly likely for these check
```

Comment:
starting from here the explanation is a bit too complex IMHO :) (it's good technical info, but not necessary to understand the tradeoff IMO, which is what matters to users I'd say)

Could we simplify the section on CPU-bound checks to something like: `For CPU-bound checks with a low activity, it's likely that the check instances will contend for the GIL, which will cause a CPU overhead compared to running these check instances sequentially (i.e. on one check runner).`